### DISCUSSION TITLE
PCA and MCR with 3D spectrum data

### DISCUSSION THREAD

**Discussion by liuchzzyy** [spectrochempy/spectrochempy#756] [Q&A] [answered]
Hi, all
Could the 3D spectral dataset be processed using PCA and MCR-ALS in spectrochempy package now?  For example, I have a dataset with x  position, y position, and spectrum for each point. 
I check the code of spectrochempy, it seems possible to do this, but not really after I tried it. right?
Thanks in advance ~
Cheng

**Comment by atravert** [comment, marked as answer]
Hi @liuchzzyy ,

MCR-ALS is intrinsically a 2-dimensional method, so to apply it to hyperspectral data (shape =n_x, n_y, n_wavenumbers) you have to:

1/  unfold the hyperspectral data in a 2-dimensional NDDataset D of the spectra (shape = n_x*n_y,  n_wavenumbers). You can either keep the spatial informations as coordinates in the NDDataset, or separately 

2/ carry out the MCR-ALS, that will yield the C (shape = nx*n_y, n_species) and St (shape = n_species, n_wavenumbers) matrices   

3/ if needed, reshape the C matrix to (n_x, n_y, n_species)

The same, I think, holds for PCA.

Does this answer your question ? 

Best regards
Arnaud

**Comment by liuchzzyy** [comment]
Hi, Arnaud
Thanks for your reply! Okay I would like to try it the way you mentioned, which is of great helps! 
and actually Yesterday I got a similar result with [tensorly.decomposition.Parafac2](https://tensorly.org/stable/modules/generated/tensorly.decomposition.Parafac2.html). Hope it useful.

Best,
Cheng

**Reply by atravert** [reply]
Yes, PARAFAC is a factor analysis method suitable for three-dimensional data sets. Please note that, like PCA, PARAFAC does not aim to recover spectral and concentration profiles (which MCR-ALS does).
Best regards,
Arnaud

**Comment by liuchzzyy** [comment]
Thanks for this info! and I am so fresh with PARAFAC.
